Publishable.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Publishable.swift
+//  pilot
+//
+//  Created by Nick Eckert on 8/28/17.
+//  Copyright © 2017 sanction. All rights reserved.
+//
+
+import Foundation
+
+import Foundation
+import Alamofire
+import AVFoundation
+
+// The publishable protocl allowes an object to be published through lightnins publish endpoint.
+protocol Publishable {
+    
+}
+
+
+extension Publishable {
+    
+    typealias SuccessHandler = () -> Void
+    typealias ErrorHandler = (Error) -> Void
+    typealias ProgressHandler = (Double) -> Void
+
+    static func publish(post: Post, with request: URLRequestConvertible, onProgress: @escaping ProgressHandler, onSuccess: @escaping SuccessHandler, onError: @escaping ErrorHandler) {
+
+        NetworkManager.sharedInstance.upload(multipartFormData: { multipartFormData in
+            switch post.postType {
```

Comment:
Are you leaving the `case .text` out for now? That would go here, right?

Reply:
Text is the default case in a post since there isn't any additional multipartFormData setup needed to perform a text based request (just a query param). Check out the Post class and you'll see the PostType variable initialized to Text.